app/src/main/java/org/kiwix/kiwixmobile/history/HistoryActivity.java

```diff
@@ -241,4 +244,15 @@ private void toggleSelection(ImageView favicon, History history) {
       actionMode.finish();
     }
   }
+
+  @Override
+  public void onBackPressed() {
```

Comment:
There are ways to exit the activity without this code triggering. Why can't we do this on click, why do we need the boolean flag?

Reply:
@mhutti1 Actually, the code is not to exit the app but to clear the stack of history when user select `Clear History` in History activity. Before what was happening is even the user selected `Clear History` option, still the activities remained open in the activity stack and were not cleared, with this code, I am just clearing those opened activities.

Boolean flag is required to decide whether to remove the opened activities or not from backstack in MainActivity.